src/qutip_cupy/cupy_dense.py

```diff
@@ -0,0 +1,36 @@
+import cupy as cp
+import numbers
+from qutip.core import data
+class CuPyDense(data.Data):
```

Comment:
This is sort of neither here nor there, but if we're going to commit to pep8, this name should be `CupyDense` (no capital p).

We could also argue that it could just be `Dense`, since it should generally be accessed through qualified access to `qutip_cupy.Dense`, so the extra bit is probably unnecessary.

Reply:
Actually I may be wrong on this - I was thinking about the "rule" that acronyms should be all capitalised.  Still `CuPyDense` looks very odd to me, because "cupy" is one word, no matter how they stylise it.  It's also rather more annoying to type!

Reply:
I will put in a small vote for `CupyDense` too, but I'm only +0 on it. I will easily live with `CuPyDense` too. :)